src/components/metadata/index.js

```diff
@@ -35,116 +52,147 @@ const MetaData = ({ visible = true, metadata, onToggleNodeSelected }) => {
-        <NodeIcon
-          className="pipeline-metadata__icon"
-          type={metadata.node.type}
-        />
-        <h2 className="pipeline-metadata__title">{metadata.node.name}</h2>
-        <IconButton
-          container={React.Fragment}
-          ariaLabel="Close Metadata Panel"
-          className="pipeline-metadata__close-button"
-          icon={CloseIcon}
-          onClick={onCloseClick}
-        />
-      </div>
-      <dl className="pipeline-metadata__list">
-        <MetaDataRow label="Type:" value={metadata.node.type} />
-        <MetaDataRow
-          label="Dataset Type:"
-          visible={isDataNode}
-          kind="type"
-          value={metadata.datasetType}
-        />
-        <MetaDataRow label="File Path:" kind="path" value={metadata.filepath} />
-        <MetaDataRow
-          label={`Parameters (${metadata.parameters?.length || '-'}):`}
-          visible={isParametersNode || isTaskNode}
-          commas={false}
-          inline={false}
-          value={metadata.parameters}
-          limit={10}
-        />
-        <MetaDataRow
-          label="Inputs:"
-          property="name"
-          visible={isTaskNode}
-          value={metadata.inputs}
-        />
-        <MetaDataRow
-          label="Outputs:"
-          property="name"
-          visible={isTaskNode}
-          value={metadata.outputs}
-        />
-        <MetaDataRow
-          label="Tags:"
-          kind="token"
-          commas={false}
-          value={metadata.tags}
-        />
-        <MetaDataRow
-          label="Pipeline:"
-          visible={Boolean(metadata.pipeline)}
-          value={metadata.pipeline}
-        />
-        <MetaDataRow
-          label="Run Command:"
-          visible={Boolean(metadata.runCommand)}>
-          <div className="pipeline-metadata__toolbox-container">
-            <MetaDataValue
-              container={'code'}
-              className={modifiers('pipeline-metadata__run-command-value', {
-                visible: !showCopied
-              })}
-              value={metadata.runCommand}
-            />
-            {window.navigator.clipboard && (
-              <>
-                <span
-                  className={modifiers('pipeline-metadata__copy-message', {
-                    visible: showCopied
-                  })}>
-                  Copied to clipboard.
-                </span>
-                <ul className="pipeline-metadata__toolbox">
-                  <IconButton
-                    ariaLabel="Copy run command to clipboard."
-                    className="pipeline-metadata__copy-button"
-                    icon={CopyIcon}
-                    onClick={onCopyClick}
+    <>
+      <MetaDataCode visible={showCodePanel} value={metadata?.code} />
+      <div className={modifiers('pipeline-metadata', { visible }, 'kedro')}>
+        {metadata && (
+          <>
+            <div className="pipeline-metadata__header-toolbox">
+              {showCodeSwitch && (
```

Comment:
With the introduction of the codeSwitch there is a need to should add / update the metadata tests to ensure the toggle button only shows up for task type nodes.

Reply:
Agreed I'll be adding some additional tests around this.